subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -161,4 +171,54 @@ public boolean isSatisfiedBy(Task element) {
 
         // TODO - check should depend on test
     }
+
+    private void configureTestedComponent(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+
+        project.getPlugins().withType(SwiftLibraryPlugin.class, new Action<SwiftLibraryPlugin>() {
+            @Override
+            public void execute(SwiftLibraryPlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+    }
+
+    private void configureTestedSwiftComponent(Project project) {
+        TaskContainer tasks = project.getTasks();
+        DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+
+        SwiftCompile compile = tasks.create("compileMainSwiftWithTestingEnabled", SwiftCompile.class, configureCompileTestedComponentTask(tasks.withType(SwiftCompile.class).getByName("compileSwift")));
```

Comment:
The test that you mentioned sounds like a functional test to me. You'll be able to test a release variant, you just won't be able to reach in and touch internals, you'll have to stick to the public APIs.

For now, we want to keep things as simple as possible and wait for feedback on which use cases people have (and don't have). There's an important difference between 'not possible' and 'not convenient'. It's certainly possible right now for a build to add its own compile task and wire that in as input to the unit test suite's link task. So the 'allow tests for my release variant to access internals' is entirely possible and supported already, it's just not particularly convenient.

We want to avoid adding complexity to the convention because someone may require a more complex wiring to satisfy their use case, as in doing so we're forcing that complexity on everyone, whether they need it or not. And that includes ourselves. Every piece we add to the convention is something we have to design, implement, test, document and keep working into the future. We want to be _very_ sure that a use case is worth the effort before we dive in. If we find out the use case with more complex wiring is required by a bunch of people, we can package that up in a plugin and people can apply the plugin, thereby layering the conventions (and complexity) to match what a particular team might need.

But that's for the future I think. For now, we should say that the debug variant of a component is unit-testable by default, and just add that flag to the compile task's compiler args. And treat the use of unit-testing your release variants as out of scope.